core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java

```diff
@@ -858,9 +897,69 @@ public String getPhysicalDestinationName() {
 		return Misc.concatStrings("url [" + url + "] mailAddress [" + (getMailAddress() == null ? "" : getMailAddress()) + "]", " ", result);
 	}
 
+	private ExchangeFileSystemResolver getResolver(String folderName){
+		return new ExchangeFileSystemResolver(folderName, getMailAddress(), getSeparator());
+	}
+
+	private ExchangeService getConnection(String mailbox) throws FileSystemException {
+		ExchangeService service = super.getConnection();
+		service.getHttpHeaders().put(ANCHOR_HEADER, mailbox);
+
+		if(!cache.isMailboxRegistered(mailbox)) {
+			try {
+				registerMailbox(mailbox, service);
+			} catch (Exception e) {
+				invalidateConnection(service);
+				releaseConnection(service);
```

Comment:
Deze release connection moet ook in een `finally {}`

Reply:
Dat lijkt mij geen goed idee, de methode heet namelijk getConnection, enkel in het geval dat er een exceptie optreedt willen we direct de connectie weer terug geven aan de pool.

Aangezien een finally {} altijd uitgevoerd wordt, ongeacht of er een exceptie optreedt, wordt de connectie al vroegtijdig terug gegeven. 

Een andere optie zou kunnen zijn een boolean waarde te introduceren en deze in het catch {} block op true te zetten en deze als conditie in het finally {} te gebruiken om te kijken of de connectie terug gegeven moet worden aan de pool.

Reply:
Ik was hier nog even naar aan het kijken en kwam eigenlijk tot de volgende conclusie:

In feite wordt overal waar deze methode aangeroepen wordt al een try-catch gedaan i.c.m invalidateConnection en releaseConnection in het finally block. 

Aangezien er een FilesystemException wordt gegeven wanneer er iets mis gaat tijdens het registreren van de mailbox zal deze connectie altijd vrijgegeven worden door de aanroepende methode.

Het try catch block lijkt dus onnodig in deze methode.

Reply:
Ik bedoel dat specifiek op deze plek de releaseConnection gedaan moet worden in een finally block, net zoals dat op regels 934-938 gedaan is.